addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,460 @@
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_STATUS_HOURS;
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_STATUS_MODE;
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_TIMER_NEXT_TIMER;
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_TIMER_STATUS;
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_VERSION_COMMENT;
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_VERSION_COMPILED;
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_VERSION_SERIAL;
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_VERSION_VERSION;
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_WLAN_SIGNAL;
+
+/**
+ * The {@link RobonectHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ * 
+ * The channels are periodically updated by polling the mower via HTTP in a separate thread.
+ *
+ * @author Marco Meyer - Initial contribution
+ */
+public class RobonectHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(RobonectHandler.class);
+
+    public RobonectHandler(Thing thing) {
+        super(thing);
+    }
+
+    private ScheduledFuture<?> pollingJob;
+    private HttpClient httpClient;
+    private RobonectClient robonectClient;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                switch (channelUID.getId()) {
+                    case CHANNEL_MOWER_NAME:
+                    case CHANNEL_STATUS_BATTERY:
+                    case CHANNEL_STATUS:
+                    case CHANNEL_STATUS_DURATION:
+                    case CHANNEL_STATUS_HOURS:
+                    case CHANNEL_STATUS_MODE:
+                    case CHANNEL_MOWER_STATUS_STARTED:
+                    case CHANNEL_TIMER_NEXT_TIMER:
+                    case CHANNEL_TIMER_STATUS:
+                    case CHANNEL_WLAN_SIGNAL:
+                        refreshMowerInfo();
+                        break;
+                    case CHANNEL_LAST_ERROR_CODE:
+                    case CHANNEL_LAST_ERROR_DATE:
+                    case CHANNEL_LAST_ERROR_MESSAGE:
+                        refreshLastErrorInfo();
+                        break;
+                    case CHANNEL_VERSION_COMMENT:
+                    case CHANNEL_VERSION_COMPILED:
+                    case CHANNEL_VERSION_SERIAL:
+                    case CHANNEL_VERSION_VERSION:
+                        refreshVersionInfo();
+                        break;
+
+                }
+            } else {
+                switch (channelUID.getId()) {
+                    case CHANNEL_MOWER_NAME:
+                        if (command instanceof StringType) {
```

Comment:
I think that nesting a if inside a switch inside an if inside a try is not nice, it could be improved by extracting both the handleRefreshCommand and the handleNormalCommands (or something like that) to their own methods. You can then leave the handling of the `RoboconnectException` and the if-else in the `handleCommand`

Reply:
ok, agree. done